src/main/java/com/avioconsulting/mule/vault/provider/api/VaultConfigurationPropertiesProvider.java

```diff
@@ -171,4 +172,33 @@ private String expandedValue(final String value) {
 
         return result;
     }
+    private void evaluateFileFallbackConfig(String fallbackFile){
+        if(fallbackFile==null || fallbackFile.isEmpty()) return;
+        try{
+            URL resourceUrl = this.getClass().getClassLoader().getResource(fallbackFile);
+            if(resourceUrl==null) return;
+            Properties appProps = new Properties();
+            appProps.load(new FileInputStream(resourceUrl.getPath()));
+            if(appProps!= null && !appProps.isEmpty())
+                fillCachedData(appProps);
+        }catch(Exception e){
+            logger.error("The follow error happened: "+ e.getMessage());
+        }
+    }
+    private void fillCachedData(Properties properties){
```

Comment:
The data is getting put into the cache, so even if a Vault instance is reachable, it won't go to Vault if the property exists in the cache when the fallback file is specified. How do you see that being handled when you only want to use the fallback file for testing?

Reply:
The fallback file is seeked in the resource folder, so if you want him to be used in test environment you put in /test/resources, if for some reason you want a fallback file in runtime you will put it in /main/resources. The file in /test/resource will not be found in runtime.